### PR TITLE
use proper lookback, and proper serialization for randomness generation

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: chain/gen/gen.go [hunk 1]
```diff
@@ -276,8 +276,12 @@ func CarWalkFunc(nd format.Node) (out []*format.Link, err error) {
...
 	mc := &mca{w: cg.w, sm: cg.sm}
 
-	// TODO: REVIEW: Am I doing this correctly?
-	ticketRand, err := mc.ChainGetRandomness(ctx, pts.Key(), crypto.DomainSeparationTag_TicketProduction, pts.Height(), m.Bytes())
+	buf := new(bytes.Buffer)
+	if err := m.MarshalCBOR(buf); err != nil {
+		return nil, nil, xerrors.Errorf("failed to cbor marshal address: %w", err)
+	}
+
+	ticketRand, err := mc.ChainGetRandomness(ctx, pts.Key(), crypto.DomainSeparationTag_TicketProduction, pts.Height(), buf.Bytes())
```
Comment (ZenGround0) [inline chain/gen/gen.go]: My reading of the spec is that the epoch number here should be `round-SPC_LOOKBACK_TICKET` which differs from this when there are null blocks.  `SPC_LOOKBACK_TICKET = 1` from spec.  This is my best reading of the spec, @sternhenri let us know if this is wrong.

Comment (whyrusleeping) [reply chain/gen/gen.go]: Yeah, good catch on that. Fixed

**Comment by whyrusleeping** [comment]
closes #1311

**Comment by Kubuxu** [comment]
Something failed in tests

**Pull request merged**